File: Week_One/Problem_Set/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    //A program that uses greedy algorithm to calculate least amount of coins owed back

    //Prompt user for change owed and Handles negative values
    float changeOwed;
    do
    {
        changeOwed = get_float("Change owed: ");
    }
    while (changeOwed < 0);

    //Converts dollars to cents
    int changeInPennies = round(changeOwed * 100);

    //Remove chunks of 25, 10, 5, 1 repeatedly while incrementing the coin total
    int coinTotal = 0;
    while (changeInPennies >= 25)
    {
        changeInPennies -= 25;
        coinTotal++;
    }

    while (changeInPennies >= 10)
    {
        changeInPennies -= 10;
        coinTotal++;
    }

    while (changeInPennies >= 5)
    {
        changeInPennies -= 5;
        coinTotal++;
    }

    while (changeInPennies >= 1)
    {
        changeInPennies -= 1;
        coinTotal++;
    }

    //Prints least amount of coins needed to give back
    printf("%i", coinTotal);
}

File: Week_6/Pset/cash/cash.py
# Implementation of cash.c in python
from cs50 import get_float

# Handle negative values
while True:
    change_owed = get_float("Change owed: ")
    if change_owed > 0:
        break

# Convert to pennies
change_in_pennies = round(change_owed * 100)

# Coin count
coin_total = 0

# Subtract Quarters and increment coin count
while change_in_pennies >= 25:
    change_in_pennies -= 25
    coin_total += 1

# Same but with Dimes
while change_in_pennies >= 10:
    change_in_pennies -= 10
    coin_total += 1

# Nickles
while change_in_pennies >= 5:
    change_in_pennies -= 5
    coin_total += 1

# Pennies
while change_in_pennies >= 1:
    change_in_pennies -= 1
    coin_total += 1

# Minimum amount of coins needed to give back
print(coin_total)